priam/src/main/java/com/netflix/priam/cluster/management/Compaction.java

```diff
@@ -0,0 +1,169 @@
+import com.netflix.priam.utils.JMXConnectorMgr;
+import org.apache.commons.lang3.StringUtils;
+import org.quartz.CronExpression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to compact the keyspaces/columnfamilies
+ * Created by aagrawal on 1/25/18.
+ */
+@Singleton
+public class Compaction extends IClusterManagement<String> {
+    private static final Logger logger = LoggerFactory.getLogger(Compaction.class);
+    private final IConfiguration config;
+    private static final Pattern columnFamilyFilterPattern = Pattern.compile(".\\..");
+    private final CassandraOperations cassandraOperations;
+
+    @Inject
+    public Compaction(IConfiguration config, IMetricPublisher metricPublisher, CassandraOperations cassandraOperations) {
+        super(config, Task.COMPACTION, metricPublisher, new CompactionMeasurement());
+        this.config = config;
+        this.cassandraOperations = cassandraOperations;
+    }
+
+    public final Map<String, List<String>> updateCompactionCFList(IConfiguration config) throws Exception {
+        final Map<String, List<String>> columnFamilyFilter = new HashMap<>(); //key: keyspace, value: a list of CFs within the keyspace
+
+        if (config.getCompactionCFList() == null || config.getCompactionCFList().isEmpty()) {
+            logger.info("Compaction: No override provided by user. All keyspaces qualify for compaction.");
+
+            if (cassandraOperations != null)
+                cassandraOperations.getKeyspaces().forEach(keyspaceName -> {
+                List<String> existingCfs = columnFamilyFilter.getOrDefault(keyspaceName, new ArrayList<>());
+                columnFamilyFilter.put(keyspaceName, existingCfs);
+            });
+            return columnFamilyFilter;
+        }
+
+        logger.info("Compaction: Override provided by user. Calculating KS/CF's to qualify for compaction");
+        String[] filters = config.getCompactionCFList().split(",");
+        for (int i = 0; i < filters.length; i++) { //process each filter
+            if (columnFamilyFilterPattern.matcher(filters[i]).find()) {
+
+                String[] filter = filters[i].split("\\.");
+                String keyspaceName = filter[0];
+                String columnFamilyName = filter[1];
+
+                if (columnFamilyName.indexOf("-") != -1)
+                    columnFamilyName = columnFamilyName.substring(0, columnFamilyName.indexOf("-"));
+
+                logger.info("Compaction: Adding CF: {}.{}", keyspaceName, columnFamilyName);
+
+                List<String> existingCfs = columnFamilyFilter.getOrDefault(keyspaceName, new ArrayList<>());
+                if (!columnFamilyName.equalsIgnoreCase("*"))
```

Comment:
In reference to your comment above, if all the CF's of a keyspace are included, we call compaction on that keyspace.